### PR TITLE
[10.0] [ADD] hr_timesheet_sheet_validators

### PR DESCRIPTION
**Pull request by nikul-serpentcs**
HR Timesheet Sheet Validators
=============================

This module allows a user outside of the Human Resources groups to validate
timesheets. A rule is predefined, but it is flexible enough to accept
extensions.

Timesheets in Odoo can be validated by users belonging to the group
“Human Resources / Officer”. However, it is frequent for companies allow to
employees outside of the Human Resources group to validate time sheet.

At the time when a user submits a time sheet to the Manager, the application
determines the validators.

Only those validators or employees in the group of Human Resources Officer
are capable of approving the timesheet.

The current rule sets as validators of a time sheet:

* The head of the department that the employee belongs to

- In case that the employee is head of the department, it will attempt to add
  the head of the parent department instead.

* The employee’s direct manager

The list of validators is visible in the employee’s timesheet.

### CODE REVIEW

**Comment by nikul-serpentcs** [comment]
@aheficent Could you please review.

File: hr_timesheet_sheet_validators/security/hr_timesheet_sheet_security.xml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+            <field name="groups" eval="[(4,ref('base.group_user'))]"/>
+        </record>
+        <record model="ir.rule" id="timesheet_validator_hr_rule">
+            <field name="name">Timesheet by HR manager</field>
+            <field name="model_id" search="[('model','=','hr_timesheet_sheet.sheet')]" model="ir.model"/>
+            <field name="global" eval="True"/>
+            <field name="domain_force">[(1,'=',1)]</field>
+            <field name="groups" eval="[(4,ref('hr.group_hr_user'))]"/>
+        </record>
+        <!-- <record id="hr_timesheet.property_rule_timesheet_employee" model="ir.rule">
```
Comment (AaronHForgeFlow) [inline hr_timesheet_sheet_validators/security/hr_timesheet_sheet_security.xml]: why this is commented?

File: hr_timesheet_sheet_validators/tests/test_hr_timesheet_sheet_validators.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+            'user_id': self.user_test.id,
+            'unit_amount': 5.00,
+        })]})
+
+    def test_timesheet_methods(self):
+        self.timesheet_sheet.action_timesheet_confirm()
+        self.timesheet_sheet.action_timesheet_done()
+        self.timesheet_sheet.action_timesheet_draft()
+        self.assertEqual(self.timesheet_sheet.validator_user_ids,
+                         self.timesheet_sheet.employee_id.parent_id.user_id)
```
Comment (AaronHForgeFlow) [inline hr_timesheet_sheet_validators/tests/test_hr_timesheet_sheet_validators.py]: can you extend the test by validating a timesheet?

**Review by AaronHForgeFlow — CHANGES_REQUESTED** [review]
Only code review

File: hr_timesheet_sheet_validators/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
+# -*- coding: utf-8 -*-
+# Copyright 2014-17 Eficent Business and IT Consulting Services, S.L.
+# Copyright 2017 Serpent Consulting Services Pvt. Ltd.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+{
+    'name': 'HR Timesheet Sheet Validators',
+    'version': '10.0.1.0.0',
+    'author': 'Eficent, Serpent Consulting Services Pvt. Ltd.',
```
Comment (LoisRForgeFlow) [inline hr_timesheet_sheet_validators/__manifest__.py]: As I said to you in a module yesterday and I am seeing that you doing some more... if you are not adding capabilities, migrations doesn't grant authorship.

**Review by LoisRForgeFlow — CHANGES_REQUESTED** [review]

File: hr_timesheet_sheet_validators/__manifest__.py (hunk 3 above)
Comment (nikul-serpentcs) [reply hr_timesheet_sheet_validators/__manifest__.py]: @lreficent please could check it.

**Review by AaronHForgeFlow — APPROVED** [review]
:+1: Thanks for attending the review

**Comment by AaronHForgeFlow** [comment]
@lreficent I think your comments have been attended

**Review by LoisRForgeFlow — APPROVED** [review]
Yes, ready to merge I think :smile:

**Pull request merged**